flink-core/src/main/java/org/apache/flink/core/memory/MemorySegment.java

```diff
@@ -279,6 +279,21 @@ public byte[] getArray() {
         }
     }
 
+    /**
+     * Returns the off heap buffer for the segment. It may be used by sink to create writer buffer
+     * for data lake such as iceberg and paimon.
```

Comment:
I tend not to emphasize that this method is used by sink for data lake. Because `MemorySegment` is a very basic class, providing method like this itself is intuitive enough.